include/config.h

```diff
@@ -13,17 +17,11 @@ typedef enum backup_type {
 typedef struct postgres_t {
 	bool enabled;
 
-	const char *origin_host;
-	const char *origin_user;
-	const char *origin_password;
-	const char *origin_port;
-	const char *origin_database;
-
-	const char *target_host;
-	const char *target_user;
-	const char *target_password;
-	const char *target_port;
-	const char *target_database;
+	d_str_t *host;
+	d_str_t *user;
+	d_str_t *password;
+	d_str_t *port;
+	d_str_t *database;
```

Comment:
Nit, maybe if we do this we won't have to add the new `CNC_MALLOC`?

```suggestion
	d_str_t host;
	d_str_t user;
	d_str_t password;
	d_str_t port;
	d_str_t database;
```